### DISCUSSION TITLE
relative line-number doesn't work in insert mode

### DISCUSSION THREAD

**Discussion by dw-rs** [helix-editor/helix#6240] [General]
the documentation states "When unfocused or in insert mode, relative will still show absolute line numbers."
so this is probably intended behavior.
however i find it quite irritating that i have to go to the normal mode, then look at the relative line numbers just so i can jump to that line in a relative manner.
i would like to be able to see the relative line number in normal mode as well so i can jump faster.
or is there a better way to jump lines in normal mode?

**Comment by pascalkuthe** [comment]
Helix is a model editor first in foremost so the intention is that movement happens in normal/visual mode. Relative line numbers are not very useful in insert-mode because you can not use counts there anyway: Pressing `2<down>` inserts 2 and then jumps down a single line.

**Reply by dw-rs** [reply]
i am well aware of the model nature of helix, and i do not have a problem with switching to normal mode to jump lines, i just want to see how many lines i need to jump while still in insert-mode, to jump faster.

**Comment by dmorawetz** [comment]
I agree with Tobias. I dislike the jumping of numbers. It would be great to have an extra configuration option (something like `always_relative`) that just shows relative numbers in every mode. I am no contributor, but from what I see it should be a simple change in https://github.com/helix-editor/helix/blob/222be0f1e769177241eb2d8d8dfb0de42e450820/helix-view/src/gutter.rs#L178-L181 and https://github.com/helix-editor/helix/blob/222be0f1e769177241eb2d8d8dfb0de42e450820/helix-view/src/editor.rs#L555-L562
Please correct me if I am wrong. If desired, I could put together a pull request.

**Reply by dmorawetz** [reply]
Created the PR #6991.
